### DISCUSSION TITLE
Leveraging Python's Asyncio for Concurrency

### DISCUSSION THREAD

**Discussion by lenardjombo** [community/community#127050] [Pull Requests]
### Select Topic Area

General

### Body

How have you utilized Python's asyncio for improving concurrency in your projects, and what challenges did you face during implementation?

**Comment by lenardjombo** [comment]
I've used **asyncio** to handle multiple network requests concurrently in a web scraping project, significantly reducing the overall runtime. The main challenge was understanding the event loop and ensuring all **async** functions were properly awaited to avoid runtime errors and potential deadlocks.

**Comment by xsxo** [comment]
if you want execute more than one function without blocking the code in the main thread use event_loop

and if you want use multi threads use create_task and run your tasks in gother function

and learn async and await

**Comment by xsxo** [comment]
i used aiohttp before

its good performance for too many requests
requests library with 2000 threads = 200 request per second
aiohttp library with 2000 tasks = 2100 request per second
